Lib/test/test_list.py

```diff
@@ -101,6 +101,28 @@ def test_empty_slice(self):
         x[:] = x
         self.assertEqual(x, [])
 
+    def test_slice_assign_iterator(self):
+        class MyIterator:
```

Comment:
Is it needed? Why `iter()` is not enough?

Reply:
It may be better to add the test in `Lib/test/list_tests.py`, as it should be tested for other list-like collections. The special case of assignment to itself is tested in `test_setslice`, so you can add a new test somewhere nearby.